src/server/game/Entities/Player/Player.cpp

```diff
@@ -12648,18 +12652,26 @@ void Player::StoreLootItem(uint8 lootSlot, Loot* loot)
 
 uint32 Player::CalculateTalentsPoints() const
 {
-    uint32 base_talent = getLevel() < 10 ? 0 : getLevel() - 9;
+    uint8 currentLevel = getLevel();
+    uint32 talentPointsForLevel = 0;
+    uint32 baseTalents = currentLevel < 10 ? 0 : currentLevel - 9;
 
     if (getClass() != CLASS_DEATH_KNIGHT || GetMapId() != 609)
-        return uint32(base_talent * sWorld->getRate(RATE_TALENT));
-
-    uint32 talentPointsForLevel = getLevel() < 56 ? 0 : getLevel() - 55;
-    talentPointsForLevel += m_questRewardTalentCount;
-
-    if (talentPointsForLevel > base_talent)
-        talentPointsForLevel = base_talent;
+    {
+        talentPointsForLevel = baseTalents;
+    }
+    else
+    {
+        talentPointsForLevel = currentLevel < 56 ? 0 : currentLevel - 55;
+        talentPointsForLevel += m_questRewardTalentCount;
+        if (talentPointsForLevel > baseTalents)
+        {
+            talentPointsForLevel = baseTalents;
+        }
+    }
```

Comment:
@kaytotes what is this logic change here for?

Reply:
Resolves https://github.com/azerothcore/azerothcore-wotlk/issues/4327 and allows custom content to easily award extra talents.

Reply:
You can close this pr if you don't keep working on it